### DISCUSSION TITLE
Prevent implicit form submission when using wizard

### DISCUSSION THREAD

**Discussion by ajnsn** [filamentphp/filament#8597] [Help]
Hey,

as usual, pressing the enter key in a text input field submits the form. Which might be handy and expectable when using single forms, it can be become an issue when using a multistep form aka wizard. 

If you have fields that fail the validation on a following step, nothing happens, no message etc. Not ideal but maybe acceptable. But if you have e.g. only a following summary step with placeholders (= without required fields), the wizard will submit immediately. The following step(s) is/are simply skipped.

One workaround might be to simply change the wizards submit action/button `type` from `submit` to `button` or to create the whole button yourself like described [in the docs](https://filamentphp.com/docs/3.x/forms/layout/wizard#rendering-a-submit-button-on-the-last-step). Then use wire:click on the button to submit the `callMountedActionName`.

Other Ideas?

**Comment by michaelklopf** [comment]
I came here because I have the same problem. Even worse, pressing enter partially saves the completed steps and you have an incomplete record in the database. Have you found a solution because you closed the issue @ajnsn?

**Comment by ajnsn** [comment]
Hey @michaelklopf 

I was using the below override of `getModalSubmitAction` in a trait. This should still work imHo. But when I learned that none of the known users press Enter, I removed it again. :)

```php

use Filament\Actions\StaticAction;

trait PreventWizardImplicitSubmit
{
    public function getModalSubmitAction(): ?StaticAction
    {
        /** @var StaticAction $action */
        $action = parent::getModalSubmitAction();

        if ($action) {
            $action->canSubmitForm = false;

            $action->extraAttributes([
                'wire:click' => $this->getLivewireCallMountedActionName(),
            ]);
        }

        return $action;
    }
}
```

**Reply by michaelklopf** [reply]
Thank you.
